### ISSUE TITLE
Esp32 servo.detach not working

### ISSUE THREAD

**Issue by StefWI** [esphome/issues#3794]
### The problem

Whether using the "servo.detach" instruction in the service section or "auto_detach_time" in the servo section or both, the servo does not detach.
The message" Servo detached on auto_detach_time" appears in the log but the servo remains impossible to move manually.

### Which version of ESPHome has the issue?

2022.10.2

### What type of installation are you using?

Home Assistant Add-on

### Which version of Home Assistant has the issue?

Home Assistant Core 2022.11.2

### What platform are you using?

ESP32

### Board

esp32dev

### Component causing the issue

servo

### Example YAML snippet

```yaml
esphome:
  name: srv-001

wifi:
  networks:
  - ssid: !secret wifi_ssid_1
    password: !secret wifi_password_1
  - ssid: !secret wifi_ssid_2
    password: !secret wifi_password_2

  # Enable fallback hotspot (captive portal) in case wifi connection fails
  ap:
    ssid: esphome.name
    password: !secret captive_portal_password

esp32:
  board: esp32dev
  framework:
    type: arduino

# Enable logging
logger:

# Enable Home Assistant API
api:
  services:
    - service: control_servo
      variables:
        level: float
      then:
        - logger.log: "Start servo change"
        - servo.write:
            id: servo_air_input
            level: !lambda 'return level / 100.0;'
        - delay: 2s
        - logger.log: "Done"
        - servo.detach: servo_air_input
        - logger.log: "Detached"
    

ota:

captive_portal:

output:
  - platform: ledc
    pin: GPIO16
    id: pwm_output
    frequency: 50 Hz
    
servo:
  - id: servo_air_input
    output: pwm_output
    auto_detach_time: 1s
```

### Anything in the logs that might be useful for us?

```txt
[16:10:38][D][main:188]: Start servo change
[16:10:38][D][servo:062]: Servo new target: 0.150000
[16:10:38][D][servo:050]: Servo reached target
[16:10:39][D][servo:028]: Servo detached on auto_detach_time
[16:10:40][D][main:041]: Done
[16:10:40][D][main:045]: Detached
```

**Comment by rdehuyss** [comment]
I've got the same problem.

**Comment by TheDukeSr** [comment]
Same problem with ESP8266. 
Servo will not detach ...;(

_Several detach code tested:_
Not working: **- servo.detach: servo_name**
Not working: **servo_name.detach();**
Not working: **auto_detach_time: 1s**

**Comment by ZonkSec** [comment]
I also have this problem on a D1 mini.

**Comment by knewg** [comment]
I have this exact problem. Has anyone managed to work around it? 

I have made a shunt controller for my acumulator tanks, but my 3D-printed mount has a bit of flex (half a mm or so) with springback, so it works great in moving the shunt to the correct position (it's pid controlled to a thermometer, so precision is not important), but when it stops moving, the flexback causes the positioning circuit to want to adjust back to the flex, but this just flexes the mount some more etc. While I could design a metal mount instead, this issue should be solved in software, since the feature exists.

Running the PWM signal through a transistor (and use it as an "enable servo movement" would solve the issue, but it's a waste of components and outputs, and worst of all, my circuit board is already done

`esphome:
  name: "tank-controller"

esp8266:
  board: d1_mini

# Enable logging
logger:

# Enable Home Assistant API
api:

ota:

servo:
  - id: shunt_level_motor
    output: shunt_level_output
    transition_length: 8s
    auto_detach_time: 2s #this does nothing, but logs it

# Example output platform
# On ESP32, use ledc output
output:
  - platform: esp8266_pwm
    id: shunt_level_output
    pin: D4
    frequency: 50 Hz


number:
  - platform: template
    id: tank_shunt_level
    name: Tank shunt level
    min_value: 0
    max_value: 100
    step: 1
    optimistic: true
    set_action:
      then:
        - servo.write:
            id: shunt_level_motor
            level: !lambda 'return x / 100.0;'
        - delay: 8s
        - servo.detach: shunt_level_motor` #This does not seem to trigger at all, but I could be doing it wrong

**Comment by paxoman** [comment]
is there any update on this problem?

**Comment by bernfp** [comment]
Confirm still the issue ESP32S3. ESPHOME 2026.3.1
